src/core/utils/context.test.ts

```diff
@@ -18,7 +18,7 @@
  */
 
 import { ContextContainer } from './context';
-import { PluginOpaqueId } from '../plugins';
+import { PluginOpaqueId } from '../server';
```

Comment:
the problem of sharing server and client code popped up several times in this PR and in https://github.com/elastic/kibana/pull/39891 
I'm wondering why we encourage feature-focus folder structure for plugins in favour of technology-focus, but do not follow the same convention in the core? why not structure core services in the similar way? we still have domain isolation
```
- core
|__ context
  |__ public
  |  |__ index.ts
  |__ server
  |  |__ index.ts
  |__ common/utils/ whatever you want
```
```

Reply:
That's not a bad idea. It could complicate some of the bundling / building logic, but I'd be interested in exploring that (in a separate PR / issue) if it helps alleviate this problem.

Reply:
could your create a separate issue to investigate this approach? we have to address problem of importing `context` from `utils` eventually.